lib/rules/amd-function-arity.js

```diff
@@ -18,7 +18,7 @@ var util = require("../util");
 
 module.exports = function (context) {
 
-    var allowExtraDependencies = (context.options && context.options[0] && context.options[0].allowExtraDependencies) || false;
+    var allowExtraDependencies = (context.options && context.options[0] && context.options[0].allowExtraDependencies) ? context.options[0].allowExtraDependencies : false;
```

Comment:
At this point, these lines are functionally identical. The `||` accomplishes the same thing as the ternary expression, in a more terse way.